client/dev-tools/components/dev-tools.tsx

```diff
@@ -33,8 +33,11 @@ const PromoCard = ( { title, text, supportContext }: PromoCardProps ) => (
 );
 
 const DevTools = () => {
-	const [ showEligibility, setShowEligibility ] = useState( false );
+	const { searchParams } = new URL( document.location.toString() );
 	const siteId = useSelector( getSelectedSiteId );
+	// The ref is required to persist the value of redirect_to after renders
+	const redirectUrl = useRef( searchParams.get( 'redirect_to' ) ?? `/hosting-config/${ siteId }` );
+	const [ showEligibility, setShowEligibility ] = useState( false );
```

Comment:
do you want to show the modal at the beginning by checking the query parameters?

Reply:
Looks it's what this PR https://github.com/Automattic/wp-calypso/pull/91226 does.